### ISSUE TITLE
Analyzer does not work with reduce method with no hashable object 

### ISSUE THREAD

**Issue by shchen-idmod** [InstituteforDiseaseModeling/idmtools#367]
### Analyzer does not work with reduce method

Current Analyzer tests PopulationAnalyzer.py and TimeseriesAnalyzer.py under idmtools_platform_comps/tests/inputs/analyzers dir does not working for reduce method

### What I Did

Right click PopulationAnalyzer,py and run in comps2. 
it will throw error:
```
Traceback (most recent call last):
  File "C:\Python37\Lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\github_ck\idmtools\idmtools_platform_comps\tests\inputs\analyzers\PopulationAnalyzer.py", line 43, in reduce
    data.group = self.default_group_fn(str(sim.uid), sim.tags)
AttributeError: 'UUID' object has no attribute 'uid'
```

This error is misleading. The main problem is caused by we do not get object back from reduce function's all_data. there is no simulation info but only uuid. that is why we are seeing error above.

Basically we need to give item as key to dict 'item_data_for_analyzer' below (item ie: DTKSimulation or PythonSimulation) in _run_and_wait_for_reducing() in AnalyzerManager.py:
item_data_for_analyzer[item_id] = item_result.get(analyzer.uid, None)  <--changed to item # item = self._items[item_id]
But there is an issue that 'item'  as key for dict is not hash-able. Do not know how to solve this issue.  This is how original dtk-tools worked. I am not sure why it is not complaining there.

```
